runtime/include/cloe/data_broker.hpp

```diff
@@ -0,0 +1,1115 @@
+  using type_erased_on_value_change_event_function_t = std::any;
+
+  /**
+    * Name(s) of the signal
+    */
+  std::vector<std::string> names_{};
+  /**
+    * std::type_info of the signal
+    */
+  const std::type_info* type_{nullptr};
+  /**
+    * getter-function
+    */
+  type_erased_get_value_function_t get_value_{};
+  /**
+    * setter-function
+    */
+  type_erased_set_value_function_t set_value_{};
+  /**
+    * Event which gets raised when the signal value changes
+    */
+  type_erased_value_changed_event_t value_changed_event_{};
+  /**
+    * Triggers the value-changed event
+    */
+  type_erased_on_value_change_event_function_t on_value_changed_{};
+  /**
+    * std::function returning the count of event subscribers
+    */
+  std::function<std::size_t()> subscriber_count_{};
+
+  /**
+    * Private default c'tor
+    * \note: Implementation-#2: The class shall be created only via a factory-method
+    */
+  Signal() = default;
+
+ public:
+  /**
+    * Public c'tor, accessible only via private access-token
+    * \note: Design-#1: The class shall be constructible via external helpers
+    */
+  explicit Signal(access_token) : Signal() {}
+  Signal(const Signal&) = delete;
+  Signal(Signal&&) = default;
+  virtual ~Signal() = default;
+  Signal& operator=(const Signal&) = delete;
+  Signal& operator=(Signal&&) = default;
+
+  /**
+   * Return the type_info for the signal
+   */
+  const std::type_info* type() const { return type_; }
+
+ private:
+  /**
+    * Validates that the stored signal type matches the template type
+    * \tparam TValue Expected type of the signal
+    */
+  template <typename TValue>
+  void validate_type() const {
+    const std::type_info* requested_type = &typeid(TValue);
+    if ((type() == nullptr) || (!(*type() == *requested_type))) {
+      throw std::logic_error(fmt::format(
+          "Type mismatch between actual and requested type. Actual type: {}. Requested type: {}",
+          type()->name(), requested_type->name()));
+    }
+  }
+
+ public:
+  /**
+    * Returns the getter function of the signal
+    * \tparam TValue Type of the signal
+    * \return Getter function of the signal
+    * 
+    * Example:
+    * ```
+    * Signal s = ...;
+    * const typed_get_value_function_t<int>* f = s.getter<int>();
+    * int v = (*f)();
+    * ```
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, const typed_get_value_function_t<TValue>*>::type getter()
+      const {
+    validate_type<TValue>();
+    const typed_get_value_function_t<TValue>* get_value_fn =
+        std::any_cast<typed_get_value_function_t<TValue>>(&get_value_);
+    return get_value_fn;
+  }
+  /**
+    * Sets the getter function of the signal
+    * \tparam TValue Type of the signal
+    * \param get_value Getter function of the signal
+    */
+  template <typename TValue>
+  typename enable_if_is_wanted<TValue, void>::type set_getter(

```

Comment:
forbid const, T*, T& -> let's name it `is_compatible`, `is_compatible_t`, `is_compatible_v`, `enable_if_compatible`, `enable_if_compatible_t`.

Put it in the databroker namespace or something?

Reply:
template <typename T>
constexpr bool is_incompatible_type_v = std::is_void_v<T> || std::is_reference_v<T>;

Done.